Client/Frontend/Fakespot/Client/FakeSpotClient.swift

```diff
@@ -0,0 +1,77 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol FakeSpotClientType {
+    func fetchProductAnalysisData(productId: String, website: String) async throws -> ProductAnalysisData
+    func fetchProductData(productId: String, website: String) async throws -> [ProductAdsData]
+}
+
+struct MockFakeSpotClient: FakeSpotClientType {
+    func fetchProductAnalysisData(productId: String, website: String) async throws -> ProductAnalysisData {
+        try load(ProductAnalysisData.self, filename: "productanalysis-response")
+    }
+
+    func fetchProductData(productId: String, website: String) async throws -> [ProductAdsData] {
+        try load([ProductAdsData].self, filename: "productadsdata-response")
+    }
+
+    private func load<T: Decodable>(_ type: T.Type, filename: String) throws -> T {
+        let path = Bundle.main.url(forResource: filename, withExtension: "json")!
+        let data = try Data(contentsOf: path)
+        return try JSONDecoder().decode(type, from: data)
+    }
+}
+
+struct StagingFakeSpotClient: FakeSpotClientType {
```

Comment:
Instead of having a client for staging we should pass in the URL for the backend (either staging or prod).